File: Data Structures & Algorithms/duplicate-integer/submission-7.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     * data structure will be set
     * for loop and to check if the set already has it already if does return true else add it to the set
     * return false whhen the loop finish
     */
    hasDuplicate(nums) {
        const isSet = new Set()
        for(const num of nums)
        {
            if(isSet.has(num)) return true;
            isSet.add(num)
            
        }
        return false
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-7.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hashNums = set(nums)
        longest = 0

        for num in hashNums:
            if (num - 1) not in hashNums:
                length = 1
                while (num + length) in hashNums:
                    length += 1
                longest = max(length, longest)
        return longest




        